jwst/pipeline/calwebb_spec2.py

```diff
@@ -336,3 +284,115 @@ def process_exposure_product(
         )
 
         return result
+
+    def _step_verification(self, exp_type, members_by_type, multi_int):
+        """Verify whether requested steps can operated on the given data
+
+        Thought ideally this would all be controlled through the pipeline
+        parameters, the desire to keep the number of config files down has
+        pushed the logic into code.
+
+        Once step and pipeline parameters are retrieved from CRDS, this
+        logic can be removed.
+        """
+
+        # Check for image-to-image background subtraction can be done.
+        if not self.bkg_subtract.skip:
+            if exp_type in WFSS_TYPES or len(members_by_type['background']) > 0:
+
+                if exp_type in WFSS_TYPES:
+                    members_by_type['background'] = []           # will be overwritten by the step
+
+                # Setup for saving
+                self.bkg_subtract.suffix = 'bsub'
+                if multi_int:
+                    self.bkg_subtract.suffix = 'bsubints'
+
+                # Backwards compatibility
+                if self.save_bsub:
+                    self.bkg_subtract.save_results = True
+            else:
+                self.log.debug('Science does not allow direct background subtraction. Skipping "bkg_subtract".')
```

Comment:
Should this be `info` level, rather than `debug`, or would that clutter up the log for too many modes that don't allow for direct background subtraction?

Reply:
Actually not sure. Guess I did debug just to keep the noise down, but in the goal of making things more explicit, and this does seem rather important info, bumping up to info sounds reasonable.

Reply:
Now that I see that there are so many of these, for all the various steps that do not apply to a given exp_type, it probably would be log overload if these were info level. So just leave them at debug.